### PR TITLE
feat: add databoost enabled property for batch transactions 

### PR DESCRIPTION
**Pull request by asthamohta**

### CODE REVIEW

**Review by gauravpurohit06 — APPROVED** [review]
LGTM

File: google/cloud/spanner_v1/database.py [hunk 1]
```diff
@@ -1135,6 +1136,11 @@ def generate_read_batches(
             service uses this as a hint, the actual number of partitions may
             differ.
 
+        :type data_boost_enabled:
+        :param data_boost_enabled:
+                (Optional) If this is for a partitioned query and this field is
```
Comment (rajatbhatta) [inline google/cloud/spanner_v1/database.py:1141]:
```suggestion
                (Optional) If this is for a partitioned read and this field is
```

File: google/cloud/spanner_v1/snapshot.py [hunk 2]
```diff
@@ -210,6 +211,14 @@ def read(
         :type timeout: float
         :param timeout: (Optional) The timeout for this request.
 
+        :type data_boost_enabled:
+        :param data_boost_enabled:
+                (Optional) If this is for a partitioned query and this field is
```
Comment (rajatbhatta) [inline google/cloud/spanner_v1/snapshot.py:216]:
```suggestion
                (Optional) If this is for a partitioned read and this field is
```

**Review by rajatbhatta — APPROVED** [review]
LGTM, with a few nit suggestions.

**Pull request merged**